test/AuthProvider.test.tsx

```diff
@@ -33,23 +36,25 @@ describe("AuthProvider", () => {
         window.history.pushState(
             {},
             document.title,
-            "/?code=__test_code__&state=__test_state__",
+            "/?code=__test_code__&state=__test_state__"
         );
         expect(window.location.href).toBe(
-            "https://www.example.com/?code=__test_code__&state=__test_state__",
+            "https://www.example.com/?code=__test_code__&state=__test_state__"
         );
 
         const wrapper = createWrapper({ ...settingsStub, onSigninCallback });
 
-        // act
-        const { waitForNextUpdate } = renderHook(() => useAuth(), {
+        renderHook(() => useAuth(), {
             wrapper,
         });
-        await waitForNextUpdate();
 
         // assert
-        expect(UserManager.prototype.signinCallback).toHaveBeenCalled();
-        expect(onSigninCallback).toHaveBeenCalled();
+        await waitFor(() => {
+            expect(UserManager.prototype.signinCallback).toHaveBeenCalledTimes(
+                2
+            );
+            expect(onSigninCallback).toHaveBeenCalledTimes(2);
```

Comment:
all these cases where `toHaveBeenCalledTimes(2)` is should be replaced with `1` when the PR #421 is merged.